### PR TITLE
port over the changes that have acumulated on the release branch

### PR DESCRIPTION
**Pull request by emberian**
Here they are!

### CODE REVIEW

**Comment by bkase** [comment]
Wow, you fixed the conflicts in one day!

**Comment by emberian** [comment]
@bkase I wouldn't go that far! I pushed the PR so it would hit CI, and apparently neglected to even build it locally :)

**Comment by bkase** [comment]
The latest:

The "Build libp2p helper" job has ever and will ever _only_ succeed if there is a cache hit with cachix. On CircleCI, in their build containers, nix builds fail with an error regarding being unable to set the hostname. As far as I can tell, this cannot be worked around (and this check has existed inside the Nix binary for at least 6 years, so it's not something that was caused by a new version).

In the short term, we should probably just populate the cache, we'll need to rethink how we build libp2p helper in the longer term, though.

File: .circleci/config.yml [hunk 1]
```diff
@@ -474,7 +474,7 @@ jobs:
                 command: LIBP2P_NIXLESS=1 GO=/usr/lib/go/bin/go make libp2p_helper
             - run:
                   name: Run unit tests
-                  command: ./scripts/skip_if_only_frontend_or_rfcs.sh bash -c 'source ~/.profile && export make build && (dune runtest src/lib --profile=dev -j8 || (./scripts/link-coredumps.sh && false))'
```
Comment (bkase) [inline .circleci/config.yml]: hopefully this fixes it!

Comment (nholland94) [reply .circleci/config.yml]: We'll see. I'm running tests locally with `-j1` so I can find the hanging test, if there is still one after this.

File: dockerfiles/Dockerfile-toolchain [hunk 2]
```diff
@@ -95,6 +95,7 @@ RUN eval $(opam env)
 
 # Install OPAM packages for spacetime compiler; use ulimited stack for camlp4 build
 ADD /src/opam.export .
+RUN opam install dune.1.11.4
```
Comment (psteckler) [inline dockerfiles/Dockerfile-toolchain:98]: dune is installed separately to allow the import to work?

Comment (bkase) [reply dockerfiles/Dockerfile-toolchain:98]: Yes -- the opam import doesn't work without this :/

**Pull request merged**